### DISCUSSION TITLE
UFO-A,B and C complete model(s)

### DISCUSSION THREAD

**Discussion by hyprdia** [nemo-ufes/gufo#40] [Q&A]
Hello, 

I would like to know where I can access the latest complete UFO ontology model(s) (A, B and C)? I would like to have access to a document similar to what we can find for [seon](https://dev.nemo.inf.ufes.br/seon/UFO.html#UFO_Social+Object_detail) but more up-to-date (this model is from 2017) and, ideally, with more complete diagrams (in the model on the [seon page](https://dev.nemo.inf.ufes.br/seon/UFO.html#UFO_Social+Object_detail) the UFO-C entities seems to be all there but they are not represented in the diagram). 

Just an access of any sort to the current models would be valuable to me as I try to implement a modelling tool based on OntoUML and UFO, but not UML-based. Even a non-official version would be useful for my project. I know the work is still ongoing, but I think it would be useful, not only for me, to have a place where we can follow the development more closely. Some libraries use a tagging scheme (stable, experimental, deprecated, etc...) to specify functions/api status, maybe something like that could be done in the UFO model for the entities!? 

It would also be quite valuable to me, to have an access of any sort to an up-to-date list of validations rules. It was first described in Dr. Giancarlo Guizzardi, Ontological Foundations for Structural Conceptual Models, but later developments probably made this list incomplete, and/or inaccurate. Now, I am following the [server code](https://github.com/OntoUML/ontouml-js) implemented for the Visual Paradigm plugin which seems quite up-to-date as it was updated last year. 

Thanks for your help,

Sébastien

P.S.1 As a side note, would it be possible to know when the official complete UFO ontology model will be released?

P.S.2 Thanks again @pedropaulofb to have directed me here.

**Comment by jpalmeida** [comment]
Sébastien, (I am sorry your question went unanswered for a very long time. I was drawn attention to it by another question posed here recently.)

Short answer is that for part A, you should look into gUFO. gUFO also has a very small fragment of UFO-B, which we are complementing soon for publication in the same style as gUFO (as an extension of it). 

About the rules, indeed, the server code is the place where they are most detailed and up to date. (However, they do not yet incorporate the rules described in https://nemo.inf.ufes.br/wp-content/papercite-data/pdf/incorporating_types_of_types_in_ontology_driven_conceptual_modeling_2022.pdf for high-order types.)

With respect to UFO-C, it is still described in the various papers (and documented in part in the SEON network), and we do not have a gUFO-based implementation. (A UML-based version exists and is maintained by Monalessa Perini Barcellos as part of her work on SEON, perhaps she would be interested in sharing this with you https://www.inf.ufes.br/~monalessa/english/ ). 

best regards,
João Paulo

**Reply by sebastienh** [reply]
Dr. Almeida (sorry in return for the late response), 

First, thank you for your response.

Since I asked the question, my work has taken me in a somewhat different direction, but your answer remains highly relevant and will likely become even more so soon.

I have continued working on UFO itself but from another angle: I am developing a dependently typed programming language grounded in UFO theory. So, your directions regarding the validation rules will come handy pretty soon. 

In fact, the paper you co-authored, "Toward a Well-Founded Theory for Multi-Level Conceptual Modeling" is one of the theoretical foundations behind this new programming language. So I would also like take the opportunity to thank you for that.

Warm regards,
Sébastien Hamel
